Model/Adapter/PurchaseVerify.php

```diff
@@ -217,22 +222,37 @@ private function _handleCustomerEmail($quote) {
             return;
         }
 
-        if (!empty($quote->getShippingAddress()->getEmail())) {
-            $quote->setCustomerEmail((string)$quote->getShippingAddress()->getEmail());
+        if (!empty($quote->getBillingAddress()) && !empty($quote->getBillingAddress()->getEmail())) {
+            $quote->setCustomerEmail((string)$quote->getBillingAddress()->getEmail());
             return;
         }
 
-        if (!empty($quote->getBillingAddress()->getEmail())) {
-            $quote->setCustomerEmail((string)$quote->getBillingAddress()->getEmail());
+        if (!empty($quote->getShippingAddress()) && !empty($quote->getShippingAddress()->getEmail())) {
+            $quote->setCustomerEmail((string)$quote->getShippingAddress()->getEmail());
             return;
         }
     }
 
+    private function _standardizeAddress($quote){
+        $billing = $quote->getBillingAddress();
+        $shipping = $quote->isVirtual() ? null : $quote->getShippingAddress();
+
+        //if not virtual but shipping email empty, use billing address email info on shipping
+        if ($shipping && empty($shipping->getEmail())) {
+            $this->logger->info('Shipping address email empty, copying from billing: '.$billing->getEmail());
+            $shipping->setEmail($billing->getEmail());
+            $quote->setShippingAddress($shipping);
+        }
+    }
+    
     private function _ignoreAddressValidation($quote) {
         $quote->getBillingAddress()->setShouldIgnoreValidation(true);
 
         if (!$quote->getIsVirtual()) {
             $quote->getShippingAddress()->setShouldIgnoreValidation(true);
+            if (!$quote->getBillingAddress()->getEmail()) {
+                $quote->getBillingAddress()->setSameAsBilling(1);
```

Comment:
Should this be

`$quote->getShippingAddress()->setSameAsBilling(1)`

Reply:
Hi Dheeraj,
This one I was following Magento's code
https://www.magentoextensions.org/documentation/class_magento_1_1_braintree_1_1_model_1_1_paypal_1_1_helper_1_1_abstract_helper.html

It says the entire function is to ensure validation success.

Although I do see your point. you reckon we should just update it?

I wanted to keep it as it is just to see if it actually does anything for our case first since there's no way to recreate the main issue